lib/OpenQA/Schema/ResultSet/Assets.pm

```diff
@@ -21,16 +21,20 @@ sub status_cache_file {
 }
 
 # called when uploading an asset or finding one in scanning
+my $INSERT = 'INSERT INTO assets (type, name, t_created, t_updated) VALUES (?, ?, now(), now()) ON CONFLICT DO NOTHING';
 sub register ($self, $type, $name, $options = {}) {
     unless ($name) { log_warning 'attempt to register asset with empty name'; return undef }
     unless (grep /^$type$/, TYPES) { log_warning "asset type '$type' invalid"; return undef }
     if (!$options->{missing_ok} && !locate_asset $type, $name, mustexist => 1) {
         log_warning "no file found for asset '$name' type '$type'";
         return undef;
     }
-    $self->result_source->schema->txn_do(
+    my $schema = $self->result_source->schema;
+    my $sth = ($self->{_insert_sth} //= $schema->storage->dbh->prepare($INSERT));
```

Comment:
Maybe you want to call `->prepare_cached` here instead and let DBI deal with the caching?